proof/refine/RISCV64/Invariants_H.thy

```diff
@@ -402,28 +652,51 @@ definition valid_ep' :: "Structures_H.endpoint \<Rightarrow> kernel_state \<Righ
    | SendEP ts \<Rightarrow> (ts \<noteq> [] \<and> (\<forall>t \<in> set ts. tcb_at' t s) \<and> distinct ts)
    | RecvEP ts \<Rightarrow> (ts \<noteq> [] \<and> (\<forall>t \<in> set ts. tcb_at' t s) \<and> distinct ts)"
 
-
-definition valid_bound_tcb' :: "machine_word option \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "valid_bound_tcb' tcb_opt s \<equiv> case tcb_opt of None \<Rightarrow> True | Some t \<Rightarrow> tcb_at' t s"
-
-definition valid_ntfn' :: "Structures_H.notification \<Rightarrow> kernel_state \<Rightarrow> bool" where
+definition valid_ntfn' :: "notification \<Rightarrow> kernel_state \<Rightarrow> bool" where
   "valid_ntfn' ntfn s \<equiv> (case ntfnObj ntfn of
-    IdleNtfn \<Rightarrow> True
-  | WaitingNtfn ts \<Rightarrow>
-      (ts \<noteq> [] \<and> (\<forall>t \<in> set ts. tcb_at' t s) \<and> distinct ts
-     \<and> (case ntfnBoundTCB ntfn of Some tcb \<Rightarrow> ts = [tcb] | _ \<Rightarrow> True))
-  | ActiveNtfn b \<Rightarrow> True)
-  \<and> valid_bound_tcb' (ntfnBoundTCB ntfn) s"
+    Structures_H.IdleNtfn \<Rightarrow> True
+  | Structures_H.WaitingNtfn ts \<Rightarrow>
+      ts \<noteq> [] \<and> (\<forall>t \<in> set ts. tcb_at' t s) \<and> distinct ts
+      \<and> (case ntfnBoundTCB ntfn of Some tcb \<Rightarrow> ts = [tcb] | _ \<Rightarrow> True)
+  | Structures_H.ActiveNtfn b \<Rightarrow> True)
+      \<and> valid_bound_tcb' (ntfnBoundTCB ntfn) s
+      \<and> valid_bound_sc' (ntfnSc ntfn) s"
```

Comment:
Please check if we can remove the `Structures_H` qualifier here. The Haskell version should be the default name space in Refine. With that, the diff for this one should be much smaller.

Reply:
Yep, we can remove them